packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -202,10 +202,27 @@ class RenderableTiledMap {
 
   /// This returns an object group fetch by name from a given layer.
   /// Use this to add custom behaviour to special objects and groups.
+  @Deprecated('This method is deprecated. Use the getLayer() method instead.')
   ObjectGroup getObjectGroupFromLayer(String name) {
     final g = map.layers.firstWhere((layer) {
       return layer is ObjectGroup && layer.name == name;
     });
     return g as ObjectGroup;
   }
+
+  /// Returns a layer of type [T] with given [name] from all the layers
+  /// of this map. If no such layer is found, null is returned.
+  T? getLayer<T extends Layer>(String name) {
+    T? foundLayer;
+
+    try {
+      foundLayer = map.layers.firstWhere((layer) {
+        return layer is T && layer.name == name;
+      }) as T;
+    } on StateError {
```

Comment:
Usually, it is not recommended to use exceptions for anything other than exceptional situations. This is because creating an exception object is a fairly inefficient process.
So, instead of a try-catch this could be rewritten as a simple for-loop, which will be simpler code too.

Reply:
This is now resolved with [70963eb](https://github.com/flame-engine/flame/pull/1405/commits/70963eb3db1c3d70dd59415b02cbcda0ce9c5eae)